### PR TITLE
fuzz: add mempool_utils.cpp

### PR DESCRIPTION
**Pull request by fanquake**
Moving the heavy (Boost) mempool code out of fuzz/util.h. Means that (for ex) a crypto_common fuzz unit doesn't need to care about seeing endless Boost headers. This results in a ~10% speedup (for me) when compiling the fuzz tests. Your results may vary.

### CODE REVIEW

File: src/Makefile.test_fuzz.include [hunk 1]
```diff
@@ -17,5 +17,6 @@ libtest_fuzz_a_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES) $(BOOST_CPPFLAGS)
 libtest_fuzz_a_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
 libtest_fuzz_a_SOURCES = \
   test/fuzz/fuzz.cpp \
+  test/fuzz/mempool_utils.cpp \
```
Comment (MarcoFalke) [inline src/Makefile.test_fuzz.include:20]: I wonder if we can nest more folders. Maybe `src/test/fuzz/util/mempool.cpp`?

File: src/test/fuzz/mempool_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+#include <limits>
+
+CTxMemPoolEntry ConsumeTxMemPoolEntry(FuzzedDataProvider& fuzzed_data_provider, const CTransaction& tx) noexcept
+{
+    // Avoid:
+    // policy/feerate.cpp:28:34: runtime error: signed integer overflow: 34873208148477500 * 1000 cannot be represented in type 'long'
+    //
+    // Reproduce using CFeeRate(348732081484775, 10).GetFeePerK()
+    const CAmount fee = std::min<CAmount>(ConsumeMoney(fuzzed_data_provider), std::numeric_limits<CAmount>::max() / static_cast<CAmount>(100000));
```
Comment (MarcoFalke) [inline src/test/fuzz/mempool_utils.cpp:20]: unrelated:
```suggestion
    const CAmount fee{ConsumeMoney(fuzzed_data_provider, /*max=*/std::numeric_limits<CAmount>::max() / CAmount{100'000})};
```

It should be possible to just pass the max. (Can do in a separate commit, if you want)

**Review by maflcko — COMMENTED** [review]
lgtm

File: src/Makefile.test_fuzz.include (hunk 1 above)
Comment (fanquake) [reply src/Makefile.test_fuzz.include]: Added some nesting.

File: src/test/fuzz/mempool_utils.cpp (hunk 2 above)
Comment (fanquake) [reply src/test/fuzz/mempool_utils.cpp]: Added another commit for this.

File: src/test/fuzz/util/mempool.h [hunk 3]
```diff
@@ -2,9 +2,12 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#ifndef BITCOIN_TEST_FUZZ_MEMPOOL_UTILS_H
-#define BITCOIN_TEST_FUZZ_MEMPOOL_UTILS_H
+#ifndef BITCOIN_TEST_FUZZ_UTIL_MEMPOOL_H
+#define BITCOIN_TEST_FUZZ_UTIL_MEMPOOL_H
 
+#include <primitives/transaction.h>
+#include <test/fuzz/FuzzedDataProvider.h>
```
Comment (MarcoFalke) [inline src/test/fuzz/util/mempool.h:9]: those two includes could be fwd-decls, but it doesn't matter because everywhere where this header is included, the full includes already exist

**Review by maflcko — APPROVED** [review]
review ACK 8a6b6dfcd8d26b62c3a13beba72440635fcc5e19 🍮

<details><summary>Show signature</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK 8a6b6dfcd8d26b62c3a13beba72440635fcc5e19 🍮
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgJuAv+MyuzOk4R07Q6iwAbpfd78pnXvv/5J2YAj5mpWWifj9ZAJL72sYk/K1I8
5K4ukFrthMelgGVV7Hkz7KcSEQu/rRpjHRRILFuYsnxGwhVYi4RCH0w+s4jrotBc
onA2qBJaft8ULZtZg1wDS+Lkl8VLQ9bwkYmUkZgJf9vaUQwMYFlO70QkqidubjcR
+lslpo9UkFD4uH8q9YLClPvGWWxpjB/6x8/CKDEVt7Yfu1cQypX6kmgrG4T2Kmgi
6FrJDAM0ECts/di/V1TLhRsGdQrSMhDxx0UawygzgaLTBc38ezG7sE17ram+ems+
cpE+mDhWu56QWG6hCrpIyRH2pkDcjymAwIjfEug9EromErGocKKpY1nK5BVaqNbl
KKhrNnZ4x+DBhvgBxCiGzfGFP2oGgDAKPY4Ke22LjdpdC1szqBcLMdWVyPCKrJBc
VzypwHcwe8rGkqZaVmGzIGB7PSEfJCEhvscA31Z7vFBsuHqTiz/OROgV+J4+/qPr
fQ8WY0tK
=GxLp
-----END PGP SIGNATURE-----
```


</details>

**Review by glozow — COMMENTED** [review]
post-merge concept ACK
